src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -281,6 +297,76 @@ public bool IsSolutionOpen
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            var isSolutionAvailable = -1;
+            var DoesNuGetSupportsAnyProject = false;
+
+            if (!IsSolutionOpen)
+            {
+                // Solution is not open. return immediately with false
+                return new Tuple<bool, bool>(false, false);
+            }
+
+            if (!DoesSolutionRequireAnInitialSaveAs())
+            {
+                // Solution is open and 'Save As' is not required. So solution is available to operate but
+                // we also need to check if solution has any supported NuGet project.
+                isSolutionAvailable = 1;
+            }
+
+            // read all deferred projects and check for packages.config based projects
+            var deferredProjects = await GetDeferredProjectsAsync();
+
+            foreach (var project in deferredProjects)
+            {
+                var vsProjectAdapter = await _vsProjectAdapterProvider.CreateAdapterForDeferredProjectAsync(project);
+
+                // project is supported in NuGet, then set DoesNuGetSupportsAnyProject
+                if (vsProjectAdapter.IsSupported)
+                {
+                    DoesNuGetSupportsAnyProject = true;
+
+                    if (isSolutionAvailable == 1)
+                    {
+                        // solution is available and also has NuGet supported project, so we don't need to go further.
+                        break;
+                    }
+
+                    // construct packages.config file path and check if it exists
+                    var packagesConfigFilePath = Path.Combine(Path.GetDirectoryName(vsProjectAdapter.FullProjectPath), "packages.config");
+                    var DoesPackagesConfigFileExist = await _vsProjectAdapterProvider.EntityExistsAsync(packagesConfigFilePath);
+
+                    if (DoesPackagesConfigFileExist)
+                    {
+                        if (isSolutionAvailable != 1)
+                        {
+                            // if there is any project with packages.config and solution is also required to be saved then solution is not available.
+                            isSolutionAvailable = 0;
+
+                            // we got the final result for both the conditions so we can break now.
+                            break;
+                        }
+                    }
+                }
+            }
+
+            // Finally, solution is open and not saved. And, only contains project.json based projects.
+            // Check if globalPackagesFolder is a full path. If so, solution is available.
+            if (isSolutionAvailable == -1)
+            {
+                var globalPackagesFolder = SettingsUtility.GetGlobalPackagesFolder(_settings.Value);
+                isSolutionAvailable = Path.IsPathRooted(globalPackagesFolder) ? 1 : 0;
```

Comment:
It's not a new code, already exists in `IsSolutionAvailableAsync` Just replicated with quick check without initialization.